### PR TITLE
Add support for deleting multiple objects

### PR DESCRIPTION
**Pull request by gregjones**
Deleting objects using label-selectors can be useful; this change
adds a new method to models allowing this to be done.

### CODE REVIEW

File: tests/k8s/test_base.py [hunk 1]
```diff
@@ -67,3 +67,15 @@ def test_find_by_name(self, client):
     def test_find_by_selectors(self, client, value, selector):
         Example.find(labels={"my_key": value})
         client.get.assert_called_once_with("/example", params={"labelSelector": selector})
+
+class TestDeleteList(object):
```
Comment (mortenlj) [inline tests/k8s/test_base.py:71]: Codacy Issue found: [expected 2 blank lines, found 1 (E302)](https://app.codacy.com/manual/fiaas/k8s/pullRequest?prid=5581357)

Comment (mortenlj) [inline tests/k8s/test_base.py:71]: Codacy Issue found: [expected 2 blank lines, found 1 (E302)](https://app.codacy.com/manual/fiaas/k8s/pullRequest?prid=5581357)

File: k8s/base.py [hunk 2]
```diff
@@ -186,6 +186,13 @@ def delete(cls, name, namespace="default", **kwargs):
         url = cls._build_url(name=name, namespace=namespace)
         cls._client.delete(url, **kwargs)
 
+    @classmethod
+    def delete_list(cls, namespace="default", labels=None, **kwargs):
+        selector = ",".join("{}{}".format(k, v if isinstance(v, LabelSelector) else Equality(v))
```
Comment (xavileon) [inline k8s/base.py:191]: I'd consider the case of no value provided (selecting on existence of a key). e.g. if value is `None` then use `Exists`. I remember needing this as the selector `key != value` would be true (so would match all objects) if key doesn't exist.

Comment (gregjones) [reply k8s/base.py:191]: So the caller can already pass in `{"foo": Exists()}` if they want that - but you think it should be a special case? The logic here is taken from `find`; I'm not sure they should be different.

**Comment by gregjones** [comment]
(Note for me: this has the commit from https://github.com/fiaas/k8s/pull/77 in it, so merge that first and rebase)

File: k8s/base.py (hunk 2 above)
Comment (xavileon) [reply k8s/base.py:191]: ah right, that's possible so that's fine 👍

**Comment by oyvindio** [comment]
I think this change in itself looks good. However, maybe it would also be a good idea to make it possible to send [DeleteOptions](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.18/#deleteoptions-v1-meta) when calling this method? I see that the existing delete single object function apparently doesn't support this either, but it probably should too https://github.com/fiaas/k8s/blob/master/k8s/base.py#L183-L187

**Comment by gregjones** [comment]
I will check what versions that's supported in; the API-docs only show it as a body-parameter since 1.14, but it might just be how they spec was rendered

File: k8s/base.py [hunk 3]
```diff
@@ -186,6 +186,13 @@ def delete(cls, name, namespace="default", **kwargs):
         url = cls._build_url(name=name, namespace=namespace)
         cls._client.delete(url, **kwargs)
 
+    @classmethod
+    def delete_list(cls, namespace="default", labels=None, **kwargs):
```
Comment (mortenlj) [inline k8s/base.py:190]: Any reason to take `kwargs` here? It's not used.

File: setup.py [hunk 4]
```diff
@@ -39,6 +39,7 @@
     "pytest-cov==2.7.1",
     "pytest-helpers-namespace==2019.1.8",
     'pytest==3.10.1',
+    'gitdb2==2.0.6',
```
Comment (mortenlj) [inline setup.py:42]: See #78

File: k8s/base.py (hunk 3 above)
Comment (gregjones) [reply k8s/base.py]: Good spot - I've changed it so now it's passed to the client call, like the other methods here

File: k8s/base.py [hunk 5]
```diff
@@ -186,6 +186,16 @@ def delete(cls, name, namespace="default", **kwargs):
         url = cls._build_url(name=name, namespace=namespace)
         cls._client.delete(url, **kwargs)
 
+    @classmethod
+    def delete_list(cls, namespace="default", labels=None, delete_options=None, **kwargs):
+        selector = ",".join("{}{}".format(k, v if isinstance(v, LabelSelector) else Equality(v))
+                            for k, v in sorted(labels.items(), key=lambda kv: kv[0]))
+        url = cls._build_url(name="", namespace=namespace)
+        if delete_options:
+            delete_options=delete_options.as_dict()
```
Comment (mortenlj) [inline k8s/base.py:195]: Codacy Issue found: [missing whitespace around operator (E225)](https://app.codacy.com/manual/fiaas/k8s/pullRequest?prid=5581357)

**Comment by gregjones** [comment]
@oyvindio the DeleteOptions as the request body does seem to work in 1.9, even though the docs don't show it, so I've added support for that

**Pull request merged**